tests/rptest/tests/tiered_storage_model.py

```diff
@@ -0,0 +1,1675 @@
+                         "Timeboxed uploads are enabled")
+
+    def make_inputs(self):
+        return [
+            # Set upload interval to very low value
+            ClusterConfigInput(
+                "EnableUploadInterval_config",
+                {"cloud_storage_segment_max_upload_interval_sec": 1},
+                restart_required=True)
+        ]
+
+
+class TS_Delete(Expression):
+    """The segment is deleted from the cloud storage"""
+    def __init__(self, model: Model = Model.default()):
+        super().__init__(model, "TS_Delete", "Bool",
+                         "Segment is deleted from the cloud storage")
+
+    def make_validators(self):
+        return [
+            # We currently don't have metric for S3 delete
+            LogBasedValidator("TS_Delete_log",
+                              "remote.*Delete object",
+                              execution_stage=TestRunStage.Intermediate),
+        ]
+
+
+class TS_STM_DeleteByGC(Expression):
+    """Segment deletion triggered by garbage collection"""
+    def __init__(self, model: Model = Model.default()):
+        super().__init__(model, "TS_STM_DeleteByGC", "Bool",
+                         "Segment deletion triggered by garbage collection")
+
+    def make_validators(self):
+        return [
+            LogBasedValidator(
+                "TS_STM_DeleteByGC_log",
+                "ntp_archiver_service.*Deleted segment from cloud storage",
+                execution_stage=TestRunStage.Intermediate),
+        ]
+
+
+class TS_STM_GarbageCollect(Expression):
+    """GC is applied to the data in the STM"""
+    def __init__(self, model: Model = Model.default()):
+        super().__init__(model, "TS_STM_GarbageCollect", "Bool",
+                         "GC is applied to the data in the STM")
+
+    def make_validators(self):
+        return [
+            LogBasedValidator(
+                "TS_STM_GarbageCollect_log",
+                # The spillover GC uses similar line that has 'spillover' after
+                # the number of segments. We need to make sure that we don't
+                # match it by using more precise pattern.
+                "ntp_archiver_service.*Deleted \d* segments from the cloud",
+                confidence_threshold=LOW_THRESHOLD,
+                execution_stage=TestRunStage.Intermediate),
+        ]
+
+
+class TS_STM_SizeBasedRetentionApplied(Expression):
+    """Size based retention is triggered in tiered-storage in the STM region of the log"""
+    def __init__(self, model: Model = Model.default()):
+        super().__init__(
+            model, "TS_STM_SizeBasedRetentionApplied", "Bool",
+            "Size based retention is triggered in tiered-storage")
+
+    def make_validators(self):
+        return [
+            LogBasedValidator(
+                "TS_STM_SizeBasedRetention_log",
+                "ntp_archiver_service.*size_based_retention.*Advancing start offset to .* satisfy retention policy",
```

Comment:
Wouldn't this catch both time and size based retention? In that case should we rename the class?